File: chapter-18-jpa-jooq/code-java/src/main/java/com/example/demo/repository/AuthorJooqRepository.java
package com.example.demo.repository;

import com.example.jooq.tables.pojos.Author;
import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.stereotype.Repository;

import java.util.Optional;

import static com.example.jooq.Tables.AUTHOR;

/**
 * 【Chapter 18】 CQRS Query(Read) 영역: jOOQ
 */
@Repository
@RequiredArgsConstructor
public class AuthorJooqRepository {

    private final DSLContext dsl;

    public Optional<Author> findById(int id) {
        return dsl.selectFrom(AUTHOR)
                  .where(AUTHOR.ID.eq(id))
                  .fetchOptionalInto(Author.class);
    }
}

File: chapter-18-jpa-jooq/code-kotlin/src/main/kotlin/com/example/demo/repository/AuthorJooqRepository.kt
package com.example.demo.repository

import com.example.jooq.tables.pojos.Author
import com.example.jooq.tables.references.AUTHOR
import org.jooq.DSLContext
import org.springframework.stereotype.Repository

/**
 * 【Chapter 18】 CQRS Query(Read) 영역: Kotlin jOOQ
 */
@Repository
class AuthorJooqRepository(
    private val dsl: DSLContext
) {
    fun findById(id: Int): Author? {
        return dsl.selectFrom(AUTHOR)
            .where(AUTHOR.ID.eq(id))
            .fetchOptionalInto(Author::class.java)
            .orElse(null)
    }
}
